jobs/demo-district-update-pass.groovy

```diff
@@ -0,0 +1,84 @@
+// Automate monthly change for PL sandbox account passwords
+// https://khanacademy.atlassian.net/browse/DIST-4092
+
+@Library("kautils")
+// Classes we use, under jenkins-jobs/src/.
+import org.khanacademy.Setup;
+// Vars we use, under jenkins-jobs/vars/.  This is just for documentation.
+//import vars.exec
+//import vars.notify
+//import vars.withTimeout
+//import vars.withSecrets
+
+new Setup(steps
+
+).allowConcurrentBuilds(
+
+).addStringParam(
+   "CYPRESS_GIT_REVISION",
+   """The name of a cypress branch to use when building.""",
+   "update-psw"
+
+).addStringParam(
+   "SLACK_CHANNEL",
+   "The slack channel to which to send failure alerts.",
+   "#demo-district-logs"
+
+).addStringParam(
+   "TEST_RETRIES",
+   """How many retry attempts to use. By default is 1.""",
+   "1"
+
+).addCronSchedule(
+   'H H 1 * *'        // Run on the first day of every month
+
+).apply()
+
+// We use the build name as a unique identifier for user notifications. 
+BUILD_NAME = "build demo-district-password-update";
+
+def _setupWebapp() {
+   kaGit.safeSyncToOrigin("git@github.com:Khan/webapp", params.CYPRESS_GIT_REVISION);
+   dir("webapp/services/static") {
+      sh("yarn install");
+   }
+}
+
+def runScript() {
+   def workDir = "webapp/services/static/javascript/districts-package/__e2e-tests__";
+   def runCypressArgs = ["yarn",
+                         "cypress",
+                         "run", 
+                         "--spec",
+                         "javascript/districts-package/__e2e-tests__/change-pswd.spec.ts",
+                         "--browser=electron",
+                         "-c {\"retries\":${params.TEST_RETRIES}, \
+                         \"screenshotOnRunFailure\":false, \"video\":false}"
+   ];
+   def dockerBuild = ["docker", "build", "-t", "update-pass", "."];
+   def dockerRun = ["docker", "run", "update-pass"];
+
+   dir('webapp/services/static') {        
+      exec(runCypressArgs); 
+   }
+
+   dir(workDir) {
+      exec(dockerBuild); 
+      exec(dockerRun); 
+   }
+}
+
+onWorker("ka-test-ec2", '4d') {
```

Comment:
```suggestion
onMaster('4d') {
```

Reply:
Thanks, I'll change it to OnMaster!

Reply:
@csilvers Seems like `OnMaster` is not [suitable](https://jenkins.khanacademy.org/job/misc/job/demo%20district/job/update-psw/41/console) for our (e2e) automation tasks, that's why we have been using the `ka-test-ec2` worker.